Use RabbitMQ for quick install guide

apaskulin opened the pull request:
<!--- Thanks for submitting a pull request! -->
<!--- If you haven't yet, please read the contributing guide at: -->
<!--- https://github.com/sensu/sensu-docs/blob/master/CONTRIBUTING.md -->
<!--- Provide a general summary of your changes in the Title above. -->

## Description
<!--- Describe your changes in detail -->
This PR:

- Revises the five-minute install guide to use RabbitMQ as the transport instead of Redis
- Splits out the Enterprise guide from the Core guide
- Removes jokes (Sorry, no fun allowed)

Tested with Vagrant and CentOS 7

To do: Copy to all versions post-review

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here: "Closes #555" -->
Promoting best practices and creating a better experience for Enterprise users

apaskulin committed:
use rabbitmq for quick install, split out enterprise

majormoses reviewed (approved):
Core content looks good, no idea on enterprise.

cwjohnston reviewed (changes requested):
In https://github.com/sensu/sensu-docs/pull/714/ we've updated our repo definitions to specify `gpgcheck=1` and provide a value for `gpgkey` attribute. 

I think we should do likewise in the repo definitions we set up here.

rgeniesse commented on content/sensu-enterprise/3.1/quick-start/five-minute-install.md:
```diff
@@ -0,0 +1,283 @@
+The Sensu Enterprise API should return a JSON array of Sensu clients similar to this:
+
+{{< highlight shell >}}
+$ curl -s http://127.0.0.1:4567/clients | jq .
```
This command is repeated from line 231. Maybe remove one? Not sure which makes the most sense.

rgeniesse commented on content/sensu-core/1.5/quick-start/five-minute-install.md:
```diff
@@ -148,52 +174,29 @@ Sensu Core users:
+The Sensu API should return a JSON array of Sensu clients similar to this:
 
 {{< highlight shell >}}
 $ curl -s http://127.0.0.1:4567/clients | jq .
```
Unrelated to these changes, but this command is repeated from line 194. Maybe remove one? Not sure which makes the most sense.

rgeniesse commented:
FWIW, I ran through all the commands in the Core and Enterprise steps. Both created working Sensu stacks. I left a comment about a repeated API command. Other than that, 💯 . Awesome work! New users should have a better experience installing Sensu.

apaskulin committed:
remove server config step

apaskulin committed:
fix gpgcheck and gpgkey

apaskulin committed:
remove duplicated command

apaskulin committed:
style consistency

apaskulin committed:
update client API output

apaskulin commented:
- Fixed @cwjohnston's request in https://github.com/sensu/sensu-docs/pull/718/commits/de09b77af28cfaf00ac54a7e0f296e439cd008fa
- Fixed @rgeniesse's request in https://github.com/sensu/sensu-docs/pull/718/commits/ea209c9f45b693d6006db0dfbd5d5f039351a01f
- Removed the server configuration step per @cwjohnston's suggestion and updated API response in https://github.com/sensu/sensu-docs/pull/718/commits/25251db7986bdf81bfe2edd3f1f164fd74cb7d4a

rgeniesse reviewed (approved):
💯 💯

apaskulin committed:
copy to all versions

apaskulin committed:
Merge branch 'master' into 5min-install

cwjohnston commented on content/sensu-core/0.29/quick-start/five-minute-install.md:
```diff
@@ -88,50 +80,70 @@ protected-mode no{{< /highlight >}}
-   "port": 4567
- }
-}' | sudo tee /etc/sensu/config.json{{< /highlight >}}
+  "rabbitmq": {
```
Sensu's RabbitMQ defaults are user `guest` password `guest` and vhost `/`. The guest user is only allowed to connect from localhost. If we want to advise folks to use non-default credentials and/or vhost, we will also need to provide the relevant rabbitmqctl commands for adding user, setting password, creating vhost and assigning permissions on that vhost to the user.

apaskulin replied:
Fixed in https://github.com/sensu/sensu-docs/pull/718/commits/2a3900e41c38f3fc190bc9e2b54ed59d52b630dc

apaskulin committed:
add vhost setup

apaskulin committed:
use consistent install method

cwjohnston reviewed (approved):
Looks good! :shipit:
